diff --git a/src/resolve/resolved-dns-dnssec.c b/src/resolve/resolved-dns-dnssec.c
index f63cd9b..cb794c9 100644
--- a/src/resolve/resolved-dns-dnssec.c
+++ b/src/resolve/resolved-dns-dnssec.c
@@ -149,7 +149,7 @@ static int dnssec_rsa_verify_raw(
         r = EVP_PKEY_verify(ctx, signature, signature_size, data, data_size);
         if (r < 0)
                 return log_debug_errno(SYNTHETIC_ERRNO(EIO),
-                                       "Signature verification failed: 0x%lx", ERR_get_error());
+                                       "Signature verification failed: 0x%u", ERR_get_error());
 
 #  pragma GCC diagnostic pop
 #else
@@ -336,7 +336,7 @@ static int dnssec_ecdsa_verify_raw(
 
         if (EC_KEY_set_public_key(eckey, p) <= 0)
                 return log_debug_errno(SYNTHETIC_ERRNO(EIO),
-                                       "EC_POINT_bn2point failed: 0x%lx", ERR_get_error());
+                                       "EC_POINT_bn2point failed: 0x%u", ERR_get_error());
 
         assert(EC_KEY_check_key(eckey) == 1);
 
@@ -361,7 +361,7 @@ static int dnssec_ecdsa_verify_raw(
         k = ECDSA_do_verify(data, data_size, sig, eckey);
         if (k < 0)
                 return log_debug_errno(SYNTHETIC_ERRNO(EIO),
-                                       "Signature verification failed: 0x%lx", ERR_get_error());
+                                       "Signature verification failed: 0x%u", ERR_get_error());
 
 #  pragma GCC diagnostic pop
 #else
@@ -512,7 +512,7 @@ static int dnssec_eddsa_verify_raw(
         evkey = EVP_PKEY_new_raw_public_key(EVP_PKEY_ED25519, NULL, key, key_size);
         if (!evkey)
                 return log_debug_errno(SYNTHETIC_ERRNO(EIO),
-                                       "EVP_PKEY_new_raw_public_key failed: 0x%lx", ERR_get_error());
+                                       "EVP_PKEY_new_raw_public_key failed: 0x%u", ERR_get_error());
 
         pctx = EVP_PKEY_CTX_new(evkey, NULL);
         if (!pctx)
@@ -532,7 +532,7 @@ static int dnssec_eddsa_verify_raw(
         r = EVP_DigestVerify(ctx, signature, signature_size, data, data_size);
         if (r < 0)
                 return log_debug_errno(SYNTHETIC_ERRNO(EIO),
-                                       "Signature verification failed: 0x%lx", ERR_get_error());
+                                       "Signature verification failed: 0x%u", ERR_get_error());
 
         return r;
 